test/libsolidity/semanticTests/array/invalid_encoding_for_storage_byte_array.sol

```diff
@@ -0,0 +1,92 @@
+contract C {
+    bytes public x = "abc";
+    bytes public y;
+    function invalidateXShort() public {
+        assembly { sstore(x.slot, 64) }
+        delete y;
+    }
+    function invalidateXLong() public {
+        assembly { sstore(x.slot, 5) }
+        delete y;
+    }
+    function abiEncode() public view returns (bytes memory) { return x; }
+    function abiEncodePacked() public view returns (bytes memory) { return abi.encodePacked(x); }
+    function copyToMemory() public view returns (bytes memory m) { m = x; }
+    function indexAccess() public view returns (byte) { return x[0]; }
+    function assignTo() public { x = "def"; }
+    function assignToLong() public { x = "1234567890123456789012345678901234567"; }
+    function copyToStorage() public { y = x; }
+    function copyFromStorageShort() public { y = "abc"; x = y; }
+    function copyFromStorageLong() public { y = "abc"; x = y; }
+    function arrayPop() public { x.pop(); }
+    function arrayPush() public { x.push("t"); }
+    function arrayPushEmpty() public { x.push(); }
+    function del() public { delete x; }
```

Comment:
How about any of these? Are they worth testing or are they just equivalent to reads and assignments?
- Assignment via storage reference.
- Passing via a reference to a library function.
- Casting to string.
- Passing data to a builtin like `keccak256()`.
- Copying as a part of a struct or an array.
- Using as a key in a mapping.

Reply:
I don't think they will use different code.